cargo-geiger/src/mapping/metadata.rs

```diff
@@ -59,7 +59,8 @@ impl MatchesIgnoringSource for cargo_metadata::Dependency {
         self.name
             == krates
                 .get_package_name_from_cargo_metadata_package_id(&package_id)
-                .unwrap()
+                .unwrap_or(package_id.to_string())
+                //.unwrap()
             && self.req.matches(
                 &krates
                     .get_package_version_from_cargo_metadata_package_id(
```

Comment:
Thanks for the fix, did you find the root cause of the problem? Why is the get_package_name_from_cargo_metadata_package_id function failing to return the package name?

@jmcconnell26 Any ideas?

Reply:
I'm not sure, @yijunyu, could you provide the package_id that fails / a link to a repository to scan that fails?
If so, I can try and write a few more test cases to catch behaviour like this.
I would like to go back at some point, and add some further error handling to this part of the code.

Reply:
I've added a print statement before the line, 
```Rust
 59         println!("package_id: {:?}", package_id.to_string());
 60         self.name
 61             == krates
 62                 .get_package_name_from_cargo_metadata_package_id(&package_id)
 63                 .unwrap()
```
It turns out that the crate that caused the error is `wasm-bindgen`. 
This can be confirmed if you run cargo-geiger with this crate at https://github.com/rustwasm/wasm-bindgen.  
I was trying to debug into this, the error is likely from `.cargo/registry/src/github.com-1ecc6299db9ec823/krates-0.5.0/src/lib.rs`, but I am not too sure about it. 
Indeed if I used the `unwrap_or_default` instead, it will skip the `wasm-bindgen` crate, which is not what we wanted.
What will be the better way to fix the error?

Reply:
Thanks for the info @yijunyu! Taking a look at the issue, something seems strange with the dependencies being added to the graph, but only when running with `wasm-bindgen` as the root of the crate. If `wasm-bindgen` is added as a dependency in a Cargo.toml, it looks to generate the report correctly.
I'll take a deeper look at this, and try to figure out what is causing the root problem.

Reply:
Thanks for looking into this: we will test it again later.

Reply:
@jmcconnell26 Any news? Just curious, no stress 😃

Reply:
@anderejd, apologies, I've been so busy lately I haven't had a chance to do half the things I've been meaning to! I may take a look at issue https://github.com/rust-secure-code/cargo-geiger/issues/175 first, and then come back to this one if you agree?

Reply:
No problem :) Take your time and thanks for #178!

Reply:
I've been trying to replicate this behaviour in a new test project, and have been stumped. I can see it happening in the `wasm-bindgen` crate, but attempting to create a new crate with the same error fails.
@anderejd, if its OK with you, I'd like to land a few PR's cleaning up the error handling, and hopefully that will lead me to the root of the problem.

Reply:
Sound great to me, go for it!